server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -439,7 +426,6 @@ protected void mergeOptions(FieldMapper other, List<String> conflicts) {
         }
         this.ignoreAbove = k.ignoreAbove;
         this.splitQueriesOnWhitespace = k.splitQueriesOnWhitespace;
-        this.fieldType().setSearchAnalyzer(k.fieldType().searchAnalyzer());
```

Comment:
Should we be able to change the `search_analyzer` like this?

Reply:
You can still update the `search_analyzer` - the `MappedFieldType` of the merged mapper is taken directly from the update, which includes `TextSearchInfo` and hence the new search analyzer.